File: Minimum number of jumps/Minimum number of jumps.py
class Solution:
    def minJumps(self, arr, n):
        #code here
        idx = 0
        d = 0
        nn = n
        while True:
            val = arr[idx]
            if val == 0: return -1
            if (len(arr) == 1): break
            temp_val = arr[idx+1:val+idx+1]
            if (idx + val >= n-1):
                d+=1
                break
            val, i = max((v+i, i) for i, v in enumerate(temp_val))

            if idx + (val-i) >= n-1:
                d+=2
                break
            else:
                idx += (i+1)
                d+=1
            
        return d

File: Minimum number of jumps/Minimum number of jumps.cpp
#include <iostream>
using namespace std;
class Solution{
  public:
    int minJumps(int arr[], int n){
        // Your code here
        int jumps =0;
        int currmax =0;
        int maxpos =0;
        for(int i=0;i<n;i++){
            if(i > maxpos){
                return -1;
                
            }
            if(maxpos >= n-1){
                return jumps;
            }
            if(currmax < i+arr[i]){
                currmax  = i+arr[i];
            }
            if(i == maxpos){
                maxpos = currmax;
                jumps++;
            }
        }
        
    }
};